src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -213,6 +213,10 @@ type KafkaAPI struct {
 	External ExternalConnectivityConfig `json:"external,omitempty"`
 	// Configuration of TLS for Kafka API
 	TLS KafkaAPITLS `json:"tls,omitempty"`
+	// TrailingDotDisabled gives ability to turn off the fully-qualified
+	// Kafka advertised DNS name.
+	// http://www.dns-sd.org/trailingdotsindomainnames.html
+	TrailingDotDisabled bool `json:"trailingDotDisabled,omitempty"`
```

Comment:
I don't think this should be setting on KafkaAPI. We use the trailing dot for other APIs as well, right?

Reply:
The bug found in the issue #1640 was specific to KafkaAPI. That's why I introduced it only in this place.

Reply:
got it. but the implementation is not aligned with the API because the headless service is used for all kinds of APIs so that change in the future will affect also FQDN on other apis.